### PR TITLE
Target C++14 standard

### PR DESCRIPTION
**Pull request by springmeyer**
Testing branch to enable c++14

### CODE REVIEW

**Comment by springmeyer** [comment]
This passed on travis except for the g++-4.8 jobs which fail, curiously, in the luabind check: https://travis-ci.org/Project-OSRM/osrm-backend/jobs/146640025#L1271

Is this worth trying to fix(the luabind check)? In other words, when we move to require or default to c++14 is there any reason to keep g++-4.8 support? /cc @TheMarex @daniel-j-h

**Comment by daniel-j-h** [comment]
gcc 4.8 is probably way too old for C++14 anyway

**Comment by TheMarex** [comment]
Yeah let's drop 4.8. Only reason we had it was for node-osrm.

**Comment by springmeyer** [comment]
Dropped g++-4.8 from testing matrix. This is rebased and ready to merge after 👀 from @TheMarex.

**Comment by daniel-j-h** [comment]
Nice! We have some gcc4.8 workarounds in the code base, would be great to rip them out.

Also what are now our requirements? gcc>=4.9, clang>=3.6?

**Comment by TheMarex** [comment]
> Also what are now our requirements? gcc>=4.9, clang>=3.6?

Yes I think gcc 4.9 will work. Not sure about clang 3.6 - was that ever a stable release?

**Comment by daniel-j-h** [comment]
Sure, comes with Ubuntu 15.04 for example.

File: .travis.yml [hunk 1]
```diff
@@ -76,19 +68,6 @@ matrix:
       compiler: "gcc-5-release-i686"
       env: TARGET_ARCH='i686' CCOMPILER='gcc-5' CXXCOMPILER='g++-5' BUILD_TYPE='Release'
 
-    - os: linux
```
Comment (TheMarex) [inline .travis.yml]: Ticket the ARM upgrade here https://github.com/Project-OSRM/osrm-backend/issues/2699

**Pull request merged**